examples/webgl-vector-tiles.js

```diff
@@ -23,40 +22,43 @@ const parsedStyleResult = parseLiteralStyle({
   'circle-fill-color': '#777',
 });
 
-class WebGLVectorTileLayer extends VectorTile {
-  createRenderer() {
-    return new WebGLVectorTileLayerRenderer(this, {
-      style: {
-        builder: parsedStyleResult.builder,
-        attributes: {
-          prop_fillColor: {
-            size: 2,
-            callback: (feature) => {
-              const style = this.getStyle()(feature, 1)[0];
-              const color = asArray(style?.getFill()?.getColor() || '#eee');
-              return packColor(color);
-            },
-          },
-          prop_strokeColor: {
-            size: 2,
-            callback: (feature) => {
-              const style = this.getStyle()(feature, 1)[0];
-              const color = asArray(style?.getStroke()?.getColor() || '#eee');
-              return packColor(color);
-            },
-          },
-          prop_strokeWidth: {
-            size: 1,
-            callback: (feature) => {
-              const style = this.getStyle()(feature, 1)[0];
-              return style?.getStroke()?.getWidth() || 0;
-            },
-          },
-        },
+// TODO: convert this style function to a flat style
+const styleFunction = createMapboxStreetsV6Style(
+  Style,
+  Fill,
+  Stroke,
+  Icon,
+  Text,
+);
+
+const style = {
+  builder: parsedStyleResult.builder,
+  attributes: {
+    prop_fillColor: {
```

Comment:
I'm confused how this matches [the expected type](https://github.com/openlayers/openlayers/blob/79c50deb6a7cc3032f43756654ddb4dc59b4b8a4/src/ol/style/flat.js#L92-L96) for `style`.

Reply:
oh right, it used to be possible to give the renderer raw shaders but I haven't open this possibility for the layer class. Sorry, my mistake!

Ideally I would convert the style for this example into a flat style like in https://github.com/openlayers/openlayers/pull/15416 but if I do that in this PR then the performance is really bad.

WDYT?

Reply:
I'm going to use a flat style instead, and once #15416 is in the performance will be acceptable again